pkg/util/proxyconfig/merge.go

```diff
@@ -19,43 +17,27 @@ import (
 // string of noProxy settings. If no user supplied noProxy settings are
 // provided, a comma-separated string of cluster-wide noProxy settings
 // are returned.
-func MergeUserSystemNoProxy(proxy *configv1.Proxy, infra *configv1.Infrastructure, network *configv1.Network, cluster *corev1.ConfigMap) (string, error) {
+func MergeUserSystemNoProxy(proxy *configv1.Proxy, infra *configv1.Infrastructure, network *configv1.Network,
+	dns *configv1.DNS, cluster *corev1.ConfigMap) (string, error) {
 	set := sets.NewString(
 		"127.0.0.1",
 		"localhost",
 	)
 
-	if len(infra.Status.APIServerURL) > 0 {
-		apiServerURL, err := url.Parse(infra.Status.APIServerURL)
-		if err != nil {
-			return "", fmt.Errorf("failed to parse api server url")
-		}
-		set.Insert(apiServerURL.Hostname())
-	} else {
-		return "", fmt.Errorf("api server url missing from infrastructure config '%s'", infra.Name)
-	}
-
-	if len(infra.Status.APIServerInternalURL) > 0 {
-		internalAPIServer, err := url.Parse(infra.Status.APIServerInternalURL)
-		if err != nil {
-			return "", fmt.Errorf("failed to parse internal api server internal url")
-		}
-		set.Insert(internalAPIServer.Hostname())
-	} else {
-		return "", fmt.Errorf("internal api server url missing from infrastructure config '%s'", infra.Name)
-	}
-
 	if len(network.Status.ServiceNetwork) > 0 {
 		set.Insert(network.Status.ServiceNetwork[0])
 	} else {
 		return "", fmt.Errorf("serviceNetwork missing from network '%s' status", network.Name)
 	}
 
+	if len(dns.Spec.BaseDomain) > 0 {
+		set.Insert("." + dns.Spec.BaseDomain)
```

Comment:
@deads2k here's where we're adding the cluster domain to noproxy. I believe we made this choice based on what @sdodson found we were doing in 3.x.

Reply:
cc @spadgett

Reply:
This would be `.cluster.local` or `.dhansen.devcluster.openshift.com`? The former is what we had added in 3.x but in 4.x since we're provisioning hosts into `.dhansen.devcluster.openshift` that zone it probably makes sense to add that as well.

Reply:
`.dhansen.devcluster.openshift.com`, i believe.  `.cluster.local` is being added via hardcode elsewhere.

Reply:
@spadgett i would expect that this routing/proxy question would have been an issue in 3.x as well, so either:

1) we were noproxying the domain in 3.x
2) customers were manually adding it
3) console traffic was going through the customer proxies without issue (which presumably means they were doing so without any additional CAs)
4) ??

Reply:
It would've been either 2 or 3.

Reply:
For 3.x, the browser talked directly to the OAuth server. We didn't make requests to the OAuth server from our pod.

The exception is the 3.11 admin console, which is the same code base as the 4.x console. We let users customize the CA for talking to the OAuth server in the admin console, though. I'm not familiar with how proxy worked in 3.x, but presumably it was possible to add HTTP proxy env vars directly to the admin console deployment.

Reply:
@sdodson @bparees that is correct, this PR uses `.${CLUSTER_NAME}.${BASE_DOMAIN}` as a default for no proxy. In my cluster, this is `.dhansen.devcluster.openshift.com`. https://github.com/openshift/cluster-network-operator/pull/295 added `.svc` and `.cluster.local` as a no proxy default.

Reply:
@danehans per the email thread discussion, let's drop the cluster domain from the noproxy autogenerated list and move forward w/ this PR.

we'll resolve the fallout (e.g. console being broken) on a case by case basis.